### ISSUE TITLE
Numix Utouch

### ISSUE THREAD

**Issue by skypce** [numixproject/numix-icon-theme-utouch#9]
Hello all, i need three icons for my free linux distribution called unidockynapse. 
I need the follow:
1.-Show-desktop icon.
2.-Windows Overview icon.
3-All windows overview icon.

How many money do you charge for this icons?
Best regards.

**Comment by tech4david** [comment]
I'd suggest you change the icon theme of your distro from uTouch to Numix Circle, as uTouch barely gets updated any more. You can find Numix Circle [here](https://github.com/numixproject/numix-icon-theme-circle). Looks like a nice distro by the way :-)

**Comment by Tetractys** [comment]
It's sad utouch are barely or not updated anymore. Utouch is the smartest
work if design ever done on an icon set in my opinion. I take care of my
Utouch installation with several tar.gz archives due to the fact they could
disappear from repository and specialized Web sites. Utouch is my preferred
icon set and it has never paired! U can offer my help to support the
project if this could be live further.
Davide Monge

**Comment by Tetractys** [comment]
Errata for my previous mail:
"I can offer my help..."

Please, consider my offer seriously. If the problem is only relative to the
time spent to maintain the package, i could do that knowing the
procedure(s) to do that. Poke me if interested :)

Il giorno lun 1 giu 2015 alle ore 13:48 Davide Monge <davide.monge@gmail.com>
ha scritto:

> It's sad utouch are barely or not updated anymore. Utouch is the smartest
> work if design ever done on an icon set in my opinion. I take care of my
> Utouch installation with several tar.gz archives due to the fact they could
> disappear from repository and specialized Web sites. Utouch is my preferred
> icon set and it has never paired! U can offer my help to support the
> project if this could be live further.
>
> Davide Monge
>

**Comment by skypce** [comment]
Hello Davide, for me too, the design of utouch icons are very beautiful, in
my opinion more than circle theme. I have a homemade linux distribution
called unidockynapse and i use in it utouch theme. I was take from
https://github.com/lbrfabio/*numix*-circle-
*blue*
I was taked the the "places" blue icons and add replace they for the yellow
utouch icons.it is only personal change.
But will be awesome if you Davide take again the utouch project. If i can
help you in something you are welcome.

*Have a great week.*


2015-06-01 8:56 GMT-03:00 Tetractys <notifications@github.com>:

**Comment by tech4david** [comment]
I'm glad you're both interested in uTouch and, with some help, maybe I could continue it. Here's some history for you first, so you can understand why it was discontinued.

I'm not _really_ part of the Numix Project, but I too liked the uTouch theme when it was discontinued by the Numix Project, so I asked if I could continue it. They said yes, and I started to update it. However, I quickly found that this could be a daunting task, because I had to manually update each icon as it was updated in the Numix Circle repo.

Then, after a quite long while, Numix released a Technic icon set, which contained a single-colour icon only of each application icon and system icon from Numix Circle and Numix Base (respectively). I then wrote a build script which would automatically turn those icons into uTouch icons, which worked - but it was quite slow and could be improved!

Anyway, I thought that the Numix Project would at least help maintain Technic too, but instead, they seemed to abandon that too, leaving me to update all the icons. This would be possible, but it could take a long time, and as I still go to school (I'm 14), and I couldn't easily do it. This is the stage we're at now, and have been for some months.

However, if both of you would like to help me continue it, that would be great! I'll see what I can do. Hopefully we can make a new version and maintain it for good.

/cc @Foggalong

**Comment by Foggalong** [comment]
I'll try to add some context to the abandonment of Technic.

One day we realised we could use a system of scripts and icon templates to automate part of the icon making process. This would allow us to launch new icon themes and bring back old ones without increasing the time spent overall.

One of these new template icon themes was Technic, designed to be used in the creation of uTouch, Shine, and Yin/Yang (the last two being themes that never reached production).

Then librsvg, the library used for handling SVG files on Linux, recieved a big update. This update completely broke all of our automation system to the point where it was beyond repair.

This meant we had to cut back the number of themes we supported back to the original cast - base, circle, and square.

**Comment by skypce** [comment]
David, do you know numix-folders? it is a script for change style and color
of icons? may be we can do it in automatic way? i dont know. I was tested
it and change the color of icons but only of "Numix" package icons.

2015-06-01 13:27 GMT-03:00 David Bailey <notifications@github.com>:

**Comment by Foggalong** [comment]
The Numix Folders script works using an entirely different scripting system. The only part of the icon creation process ut handles is the custom recolouring, everything else is just prerendered icons.

**Comment by Tetractys** [comment]
To: all
I see the point and the objective difficulties. Despite i'm an old user of Linux (from 1994) and i have some "agility" with scripts, system management etc. i never approached the mechanics behind icon packaging/managing/creation. However i'm willing to learn so, if there is some step procedure to follow i'll do it. I have a good machine (i7 3.6GHz) and all cpu and man time to spare. Consider me as a low level but tireless worker (i'll learn by doing).

To Foggalong:
you wrote:
......
Then librsvg, the library used for handling SVG files on Linux, recieved a big update. This update completely broke all of our automation system to the point where it was beyond repair.
......

Q: any chance to review the code or this would be rewrite it from scrach?

To tech4david:
As you can read above, here i am - i think we can try!

To skypce:
Hi mate :) i think we have at least reopened the discussion. uTouch, as the other ones are a result of a big work (beside the facts that all are awesome icons) so, hoping we can help in some manner.

David

**Comment by Foggalong** [comment]
@Tetractys It'd be a case of completely rewriting the scripts as far as I know. The breakage was pretty immense, but unfortunately it's not something we have the time to do in an official capacity :(

**Comment by Tetractys** [comment]
@Foggalong so, let try to do something as "unofficial" capacity :) at least i could read the old librsvg definitions as well as the old script (in read only mode eventually - avoiding the creation of a fork for the moment) and read about the new modifications in actual librsvg. If i understand the entity of the work then maybe i'll can, with your supervision, try to do the work... at least some.
What do you think about?

**Comment by Foggalong** [comment]
@Tetractys That would be awesome... if we still had the scripts. Unfortunately this was all about a year ago so we don't have them any more :(

**Comment by tech4david** [comment]
@Foggalong I do still have my dodgy scripts, at tech4david/utouch-build-scripts-old

**Comment by Tetractys** [comment]
@tech4david i just saw  the scripts at tech4david/utouch-build-scripts-old. They seems pretty readable bash and quite understandable to me. Can this be a starting point? i mean the "svgcleaner" and all connected dev libs in the new version, as Foggaiong said, have for sure different and upgraded mode to access them....

**Comment by tech4david** [comment]
I think that in terms of the program logic, it could be a starting point, but the code itself should be rewritten. It's readable, just repetitive and inefficient, which makes it kinda slow and hard to update. Also, if I do it again I might make a real Makefile, or write the script(s) completely in Ruby or Python alternatively which might make it easier.

**Comment by Foggalong** [comment]
If someone does indeed take up uTouch, Numix are more that happy to provide the infrastructure for it. Place in the repos, website, etc

**Comment by tech4david** [comment]
@Foggalong I've still got full access to the Numix repository ;) just haven't done much because of reasons mentioned above

**Comment by skypce** [comment]
Do you allow me use numix uTouch in my linux distro? really i only was
taked it from repositories and apply to him without ask by the owners.

2015-06-01 15:54 GMT-03:00 David Bailey <notifications@github.com>:

**Comment by tech4david** [comment]
Yeah, sure you can use it. Just try to give attribution somewhere if possible :) maybe @Foggalong can give you more info?

**Comment by skypce** [comment]
thank you so much, David. i will add in my features list "Numix uTouch
icons provide for Numix team"

2015-06-01 16:02 GMT-03:00 David Bailey <notifications@github.com>:

**Comment by Tetractys** [comment]
@Foggalong @tech4david let me talk with a friend that could help me in this and the word "indeed" will become a reality :)

**Comment by tech4david** [comment]
Have a look at this everyone :) https://github.com/tech4david/utouch

Only one icon is included for demonstration purposes. This is basically a much better rewrite of my original build scripts.

Note that (at the moment) it contains the Technic icons in its own repository, but we might be able to change this later for convenience.

**Comment by tech4david** [comment]
:arrow_up: In Ruby of course :D

**Comment by Tetractys** [comment]
@tech4david i'm about to talk with one of my friend to offer a solid presence if we chose to "indeed" sustain the project. So, just wait a day or two and you all will have our communtications. i will take a look at https://github.com/tech4david/utouch with my friend. We want to embrace this project seriously so we are analyzing all the present material available on Github. stay in touch! prpbably we will have some (for sure! :) ) questions.

**Comment by Tetractys** [comment]
@tech4david @skypce @Foggalong @stefanomad
Tech4david, me and @StefanoMad are are ready to deeply analyze the environment. What we could do, for clarity and speed, is to have a Google Hangouts in audio or a skype session to have an overview from you or one of the other maintainers. Seems sto me the fastest method to have an overview of the environment.
Let we know asap so we can start on!

**Comment by StefanoMad** [comment]
Hello all, with @Tetractys we era going to analyze and write what is needed to be writen to make this awesome set and theme works with the new librsvg version. We already taken a look on the ruby scripts done by @tech4david and found some hits on what change from the "old" icon set... stay tuned :)
